### PR TITLE
Add generic checkov workflow

### PR DESCRIPTION
**Pull request by andyloughran**
Added an externally called checkov workflow for executing checkov and uploading the results to GHAS using sarif.

### CODE REVIEW

**Review by paroxp — APPROVED** [review]

**Comment by paroxp** [comment]
With such setup, we're running the linter after merge. Is that sufficient, to have a followup PRs fixing the linter, or would we benefit more from a pre-merge check?

File: workflow-templates/checkov.yml
Comment (andyloughran) [inline workflow-templates/checkov.yml:12]: Outstanding ticket next needs to be to change this to a release process e.g. `@v1` instead of `@main` before it gets widely used.

Comment (cjfryer) [reply workflow-templates/checkov.yml:12]:
```suggestion
    uses: govuk-one-login/.github/.github/workflows/checkov.yml@e3caa6e3d5fc7b3794bec3c00f9d8db1e7446888 # v0
```

Does this satisfy the requirement, if we tag this merge commit as version 0?

Comment (andyloughran) [reply workflow-templates/checkov.yml:12]: @cjfryer if you're happy with that approach, dismiss this review and merge and call me and we'll do a follow up merge.

**Pull request merged**
